app/renderer/components/publisherToggle.js

```diff
@@ -23,97 +24,89 @@ class PublisherToggle extends ImmutableComponent {
     this.onAuthorizePublisher = this.onAuthorizePublisher.bind(this)
   }
 
-  get publisherId () {
-// @cezaraugusto: publisherIds aren't limited to a domain name
-// when this runs, do you think that app/extensions/brave/content/scripts/pageInformation.js has already run?
-
-    return tldjs.getDomain(this.props.url)
+  get locationId () {
+    return getBaseUrl(this.props.location)
   }
 
-  get hostPattern () {
-    return `https?://${this.publisherId}`
+  get publisherId () {
+    return this.props.locationInfo.getIn([this.locationId, 'publisher'])
   }
 
   get hostSettings () {
-    // hostPattern defines it's own identifier for authorized publishers
-    // sites that do not match criteria would populate siteSettings
-    // with their default protocol, not hostPattern
-    return this.props.hostSettings.get(this.hostPattern)
+    const hostPattern = getHostPattern(this.publisherId)
+    return this.props.siteSettings.get(hostPattern)
   }
 
   get validPublisherSynopsis () {
-    // If session is clear then siteSettings is undefined and icon will never be shown,
-    // but synopsis may not be empty. In such cases let's check if synopsis matches current publisherId
-    return this.props.synopsis.map(entry => entry.get('site')).includes(this.publisherId)
+    // If session is clear then siteSettings is undefined and icon
+    // will never be shown, but synopsis may not be empty.
+    // In such cases let's check if synopsis matches current publisherId
+    return !!this.props.synopsis.map(entry => entry.get('site')).includes(this.publisherId)
   }
 
-  get authorizedPublisher () {
-    // If we can't get ledgerPayments, then it's likely that we are
-    // on a clean session. Let's then check for publisher's synopsis
+  get enabledForPaymentsPublisher () {
+    // All publishers will be enabled by default if AUTO_SUGGEST is ON,
+    // excluding publishers defined on ledger's exclusion list
+    const excluded = this.props.locationInfo.getIn([this.locationId, 'exclude'])
+    const autoSuggestSites = getSetting(settings.AUTO_SUGGEST_SITES)
+
+      // hostSettings is undefined until user hit addFunds button.
+      // For such cases check autoSuggestSites for eligibility.
     return this.hostSettings
       ? this.hostSettings.get('ledgerPayments') !== false
```

Comment:
Do all hostSettings entries have a ledgerPayments field? because comparing !== false will be true if that field is undefined.  you might want to change this to something like:
```
return this.hostSettings && typeof this.hostSettings.get('ledgerPayments') === 'boolean'
    ? this.hostSettings.get('ledgerPayments') !== false
    : this.validPublisherSynopsis || (autoSuggestSites && !excluded)
```